lib/src/data/communication/api/api_errors.dart

```diff
@@ -0,0 +1,33 @@
+abstract class ApiError {
```

Comment:
Zastanawiam się, że może fajnie jakby dziedziczyło po exception?

Reply:
oprócz nazewnictwa i przeznaczenia coś więcej nam to da? btw możemy ew implementować Exception, nie da się po tym dziedziczyć

Reply:
taka pierwsza myśl była taka że jak masz błąd to reprezentowany jest jako exception, takie idiomatyczne mi się to po prostu wydaje. Też nie do końca wiem jak obsługa błędów będzie wyglądać ale może to dać tyle, że pozwoli na obsługę tak jak język sobie radzi z obsługą exceptionów jak choćby taki catch na obiektach typu future do przechwycenia i obsłużenia wyjątku